test/blackbox-tests/export-invalid-names.t

```diff
@@ -0,0 +1,47 @@
+Export otherwise invalid OCaml identifiers to JavaScript
+
+  $ . ./setup.sh
+  $ cat > dune-project <<EOF
+  > (lang dune 3.8)
+  > (using melange 0.1)
+  > EOF
+  $ cat > dune <<EOF
+  > (melange.emit
+  >  (compile_flags -w -32)
+  >  (emit_stdlib false)
+  >  (preprocess (pps melange.ppx))
+  >  (target js-out))
+  > EOF
+
+  $ cat > x.ml <<EOF
+  > let [@mel.as POST] post= 1
+  > let [@mel.as "GET"] get = 2
+  > let [@mel.as "put"] other = 2
+  > let [@mel.as "class"] other = 2 (* class is reserved in JS *)
+  > EOF
+
+  $ cat > x.mli <<EOF
+  > val [@mel.as "GET"] get : int
+  > val [@mel.as "class"] other : int
+  > EOF
+
+  $ dune build @melange
+
+  $ cat _build/default/js-out/x.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  var get = 2;
+  
+  var other = 2;
+  
+  var GET = 2;
+  
+  var $$class = 2;
```

Comment:
I haven't seen the $$prefix logic in this diff. WDYT about raising instead of appending $$?

Reply:
That would

1) be a breaking change
2) not allow you to name any OCaml functions with reserved JS names, therefore invalidating a lot of OCaml code out there.